cmake_files/dependencies.cmake

```diff
@@ -1,9 +1,5 @@
 include(PackageLookup)  # check for existence, or install external projects
 
-lookup_package(Eigen3 REQUIRED ARGUMENTS MD5 9e30f67e8531477de4117506fe44669b URL https://gitlab.com/libeigen/eigen/-/archive/3.3.7/eigen-3.3.7.tar.gz)
-if(logging)
-  lookup_package(spdlog REQUIRED)
-endif()
```

Comment:
The package installation is being handled by conan but I would think we'd still need to call `find_package()` for these + FFTW3 below.

The tests build and pass currently with these lines removed, which is a bit strange.

TODO: Build with regressions=on and see if we get an error about FFTW3
TODO: Build with logging=on
TODO: Add `find_package(Eigen3)` and `find_package(spdlog)`

Reply:
Switching `CMAKE_VERBOSE_MAKEFILE=ON` again, I can see that conan inserts the paths to all the dependencies it has installed to the compiler flags directly. A compile command looks something like

```
[ 76%] Building CXX object cpp/examples/CMakeFiles/example_l1_proximal.dir/l1_proximal.cc.o
cd /Users/tkoskela/git_repos/sopt/build/cpp/examples && ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++  -I/Users/tkoskela/.conan/data/eigen/3.3.7/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/eigen3 -I/Users/tkoskela/.conan/data/catch2/2.13.7/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include -I/Users/tkoskela/.conan/data/benchmark/1.6.0/_/_/package/9e63fd2936c0b4a43222b423ad03ee6f6975cbfb/include -I/Users/tkoskela/.conan/data/libtiff/4.0.9/_/_/package/9db7415868fa23bc26464403e5de51da02c50878/include -I/Users/tkoskela/.conan/data/spdlog/1.9.2/_/_/package/2c8ec9b22128f36867b60dd65efb316eb93c9155/include -I/Users/tkoskela/.conan/data/zlib/1.2.11/_/_/package/e4fbacdee83eb1a2e68227ffe19f6e9a61b36cbc/include -I/Users/tkoskela/.conan/data/xz_utils/5.2.5/_/_/package/e4fbacdee83eb1a2e68227ffe19f6e9a61b36cbc/include -I/Users/tkoskela/.conan/data/libjpeg/9d/_/_/package/e4fbacdee83eb1a2e68227ffe19f6e9a61b36cbc/include -I/Users/tkoskela/.conan/data/jbig/20160605/_/_/package/004c156104209a65ece6cc9d71d8115366dd9d08/include -I/Users/tkoskela/.conan/data/fmt/8.0.1/_/_/package/db14d83fb6d5b354ad734851c8c78016652c0ce0/include -I/Users/tkoskela/git_repos/sopt/cpp/sopt/.. -I/Users/tkoskela/git_repos/sopt/build/include -isystem /usr/local/include -isystem /usr/local/Cellar/open-mpi/4.1.2/include -std=c++11   -O3 -DNDEBUG  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -DLIBJPEG_STATIC -DLZMA_API_STATIC -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL "-Xclang -fopenmp" -MD -MT cpp/examples/CMakeFiles/example_l1_proximal.dir/l1_proximal.cc.o -MF CMakeFiles/example_l1_proximal.dir/l1_proximal.cc.o.d -o CMakeFiles/example_l1_proximal.dir/l1_proximal.cc.o -c /Users/tkoskela/git_repos/sopt/cpp/examples/l1_proximal.cc
```

So do I understand correctly then that we do not need to call `find_package()` in cmake at all? @dpshelio

Reply:
Built on Ubuntu with logging=on builds fine, tests 1-21 pass, test 22 took too long so I killed it but will try again later